pkg/authenticator/config/config.go

```diff
@@ -28,91 +29,94 @@ const (
 	DefaultClientCertPath = "/etc/conjur/ssl/client.pem"
 	DefaultTokenFilePath  = "/run/conjur/access-token"
 
+	DefaultConjurVersion = "5"
+
 	// DefaultTokenRefreshTimeout is the default time the system waits to reauthenticate on error
 	DefaultTokenRefreshTimeout = 6 * time.Minute
 )
 
-// New returns a new authenticator configuration object
+var requiredEnvVariables = []string{
+	"CONJUR_AUTHN_URL",
+	"CONJUR_ACCOUNT",
+	"CONJUR_AUTHN_LOGIN",
+	"MY_POD_NAMESPACE",
+	"MY_POD_NAME",
+}
+
+// ReadFileFunc defines the interface for reading an SSL Certificate from the env
+type ReadFileFunc func(filename string) ([]byte, error)
+
+// NewFromEnv returns a config FromEnv with the default functional options
 func NewFromEnv() (*Config, error) {
-	var err error
+	return FromEnv(ioutil.ReadFile)
+}
+
+// FromEnv returns a new authenticator configuration object
+func FromEnv(readFileFunc ReadFileFunc) (*Config, error) {
+	var err error = nil
```

Comment:
Can drop the nil, since that's the zero value for an error type, i.e. just use `var err error`.